CHANGELOG.md

```diff
@@ -13,10 +13,24 @@ For Helidon 1.x releases please see [Helidon 1.x CHANGELOG.md](https://github.co
 
 ## [4.3.0-SNAPSHOT]
 
-This minor release of Helidon contains important bugfixes and enhancements and is recommended for all users of Helidon 4. It is API compatible with Helidon 4.2.X.
+This minor release of Helidon contains important bugfixes and enhancements and is recommended for all users of Helidon 4.
 
 A minimum of Java 21 is required to use Helidon 4.
 
+### BREAKING CHANGES
+
+- `io.helidon.common.config.Config` is deprecated in favor of `io.helidon.config.Config`.
```

Comment:
The deprecation of `io.helidon.common.config.Config` in favor of `io.helidon.config.Config` is not actually a breaking change, it could be documented in a different section (DEPRECATIONS?).

We need to make this easy to understand for end-users, right now it's confusing.
Maybe @barchetta can help.